tree/ntuple/v7/src/RPageStorageDaos.cxx

```diff
@@ -100,8 +100,21 @@ RDaosURI ParseDaosURI(std::string_view uri)
       throw ROOT::Experimental::RException(R__FAIL("Invalid DAOS pool URI."));
    return {m[1], m[2]};
 }
-} // namespace
 
+std::pair<uint32_t, uint32_t> DecodeDaosPagePosition(int64_t address)
+{
+   uint32_t position = static_cast<uint32_t>(address & 0xFFFFFFFF);
+   uint32_t offset = static_cast<uint32_t>(address >> 32);
+   return {position, offset};
+}
+
+int64_t EncodeDaosPagePosition(uint64_t position, uint64_t offset = 0)
+{
+   int64_t address = static_cast<int64_t>(position);
```

Comment:
I'd prefer to perform bitwise operations on unsigned integers, as that avoids problems with sign extension, etc.